addons/mail/models/ir_attachment.py

```diff
@@ -67,14 +67,15 @@ def register_as_main_attachment(self, force=True):
         if not self.res_model or not self.res_id:
             return
         related_record = self.env[self.res_model].browse(self.res_id)
-        if not related_record or \
-                not related_record.check_access_rights('write', raise_exception=False) or \
-                not hasattr(related_record, 'message_main_attachment_id'):
+        if not hasattr(related_record, '_message_set_main_attachment_id'):
             return
 
-        if force or not related_record.message_main_attachment_id:
-            with contextlib.suppress(AccessError):
-                related_record.message_main_attachment_id = self
+        try:
+            related_record._message_set_main_attachment_id(self.ids, force=force)
+        except AccessError:
```

Comment:
Wasn't that what "contextlib.suppress" was doing?

Reply:
It is back anyway, batch mode works best with suppress :)